File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.py
class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        ans = ''

        for word in words:
            is_palindrome = True
            length = len(word)//2
            for i in range(length):
                if word[i] != word[-i-1]:
                    is_palindrome = False
                    break
            if is_palindrome == True:
                ans = word
                break

        return ans

File: 2108-find-first-palindromic-string-in-the-array/2108-find-first-palindromic-string-in-the-array.cpp
class Solution {
public:
    string firstPalindrome(vector<string>& words) {
        string ans;

        for (auto word:words) {
            bool is_palindrome = true;
            int length = word.size();
            for (int i=0;i<length/2;i++) {
                if (word[i] != word[length-i-1]) {
                    is_palindrome = false;
                    break;
                }
            }
            if (is_palindrome == true) {
                ans = word;
                return ans;
            }
        }           
        return ans;
    }
};